modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/MinusNode.java

```diff
@@ -112,6 +112,14 @@ public void end(int idx) throws Exception {
 
         checkState();
 
+        if (type == AggregateType.SINGLE && grouping.isEmpty()) {
+            requested = 0;
+
+            downstream().end();
```

Comment:
Subsequent ending of downstream is unwanted because you have to be sure a caller side is waiting for next row. Currently it will be violated in case underlying source is empty, and caller invokes `request(0)`. This case is syntetic right now, yet still possible.

Reply:
Fixed.